### PR TITLE
Fix shadow bleeding at shorelines

### PR DESCRIPTION
**Pull request by daleeidd**
Jittering shadow sampling causes shadows to bleed as a sample can land inside of geometry where it is shadowed. Easier to explain with screenshots:

You may notice there is still leaking, but that is only because the data being rendered is from the vertex shader which doesn't have enough density at that distance. Below is if the shadows are moved to the pixel shader (not part of this PR):

## Performance

Adds about 0.04-0.05ms to the entire shadow sampling pass.

## Testing

I have left the shadows rendered on the ocean surface on (from the vertex shader). Toggle "Create Sea Floor Depth Data" to see before and after.

### CODE REVIEW

File: crest/Assets/Crest/Crest/Shaders/Ocean.shader [hunk 1]
```diff
@@ -438,6 +438,8 @@ Shader "Crest/Ocean"
 				// We need this when sampling a screenspace texture.
 				UNITY_SETUP_STEREO_EYE_INDEX_POST_VERTEX(input);
 
+				return half4(input.flow_shadow.zw, 0.0, 1.0);
```
Comment (huwb) [inline crest/Assets/Crest/Crest/Shaders/Ocean.shader:441]: Assume this is debug :)

**Review by huwb — APPROVED** [review]
Yeah I think this works. Trying to think of it will fail in some situation. With checking if it works inside the cave in pirate cove.

Approved anyway, nice work

File: crest/Assets/Crest/Crest/Shaders/Ocean.shader (hunk 1 above)
Comment (daleeidd) [reply crest/Assets/Crest/Crest/Shaders/Ocean.shader]: Indeedy. Removed :^)

**Comment by daleeidd** [comment]
> Yeah I think this works. Trying to think of it will fail in some situation. With checking if it works inside the cave in pirate cove.
> 
> Approved anyway, nice work

Thanks. I checked Pirate Cove and it has no negative affect. If I add the rocks to the depth cache it improves it. I checked different sun orientations too. Overall an improvement.

**Pull request merged**